sprockets_amqp/client.py

```diff
@@ -0,0 +1,497 @@
+import logging
+import time
+import uuid
+
+from tornado import concurrent, ioloop
+import pika.exceptions
+
+from sprockets_amqp import exceptions
+
+
+STATE_IDLE = 1
+STATE_CONNECTING = 2
+STATE_READY = 3
+STATE_BLOCKED = 4
+STATE_CLOSING = 5
+STATE_CLOSED = 6
+STATE_DESC = {STATE_IDLE: 'Idle',
+              STATE_CONNECTING: 'Connecting',
+              STATE_READY: 'Ready',
+              STATE_BLOCKED: 'Blocked',
+              STATE_CLOSING: 'Closing',
+              STATE_CLOSED: 'Closed'}
+
+DEFAULT_RECONNECT_DELAY = 5
+DEFAULT_CONNECTION_ATTEMPTS = 3
+
+
+class Client(object):
+    """This class encompasses all of the AMQP/RabbitMQ specific behaviors.
+
+    If RabbitMQ closes the connection, it will reopen it. You should
+    look at the output, as there are limited reasons why the connection may
+    be closed, which usually are tied to permission related issues or
+    socket timeouts.
+
+    If the channel is closed, it will indicate a problem with one of the
+    commands that were issued and that should surface in the output as well.
+
+    """
+    STATE_DESC = STATE_DESC
```

Comment:
I don't think anything is using this but the class for ``self.state_description`` -- why keep it?